### DISCUSSION TITLE
Is it possible to query remote in the watchOne only if local record is not found?

### DISCUSSION THREAD

**Discussion by andreystavitsky** [flutterdata/flutter_data#232] [Q&A] [answered]
For example, I have chat page with list of rooms, each list tile contains subtitle with avatar and username of author of the last message. If I use the consumer widget that will watch message author - it will make query each time the message changes. If I use watchOne with remote == false, it will not query users api if there's no local one saved.

**Comment by frank06** [comment, marked as answer]
Override your adapter:

```
DataStateNotifier<Author?> watchOneNotifier() {
  // use findOne local to check and remote to fetch in the background if necessary
  // localAdapter.findOne(key); for sync API
  // call super.watchOneNotifier()
}
```
